### DISCUSSION TITLE
steps for docker-compose + localhost were a bit unclear for me

### DISCUSSION THREAD

**Discussion by bpmct** [pomerium/pomerium#2230] [Ideas]
Just a bit of feedback on my setup journey, I ultimately got it working and this may be more clear to others:

I was a bit confused when following the [quick start guide for docker-compose](https://www.pomerium.io/docs/quick-start/). I saw it used a localhost proxy, and that made sense to me, but I wasn't sure if I needed to use autocert or not.

I assumed autocert would probably fail with the localhost URL, so I went the [self-signed wildcard certificate](https://www.pomerium.io/docs/topics/certificates.html#certificates-2) route. Edit: I tested afterwards, and yes it did fail.

This worked as expected, except for a little hiccup where I needed to install mkcert with homebrew instead.

After that though, it took a bit more searching on how to ultimately mount those certificates, AND add to `config.yaml`. Given the example uses the localhost URL, I think it would make sense to add a bit more detail to that. It kind of felt like I was diverging from the guide and doing something wrong since I had to find two additional guides. Maybe I was?

I ultimately got it working though:
https://github.com/bpmct/pomerium-code-server

**Comment by bpmct** [comment]
I'm happy to submit a PR to make these fixes, but I wanted to create this discussion first to make sure I set it up as intended.

**Reply by desimone** [reply]
> I was a bit confused when following the quick start guide for docker-compose. I saw it used a localhost proxy, and that made sense to me, but I wasn't sure if I needed to use autocert or not.
> I assumed autocert would probably fail with the localhost URL, so I went the self-signed wildcard certificate route. Edit: I tested afterwards, and yes it did fail.



You are right. Given that this is a "quick start" we should probably just start people off with (or link to) the mkcert / wildcard certificates. Not specifying this up front is probably a point of confusion. 

We should probably just give the `mkcert` instructions, and link to the more general TLS instructions for folks to look at as they advance. 

> After that though, it took a bit more searching on how to ultimately mount those certificates, AND add to config.yaml. Given the example uses the localhost URL, I think it would make sense to add a bit more detail to that. 

Agree. We should provide people with the exact configuration steps to get the most basic version of Pomerium version as part of the quick start. 

> It kind of felt like I was diverging from the guide and doing something wrong since I had to find two additional guides. Maybe I was?

No -- the err is on our part. At least for the quick-starts, the goal is to get users from 0-something to play with in as little time as possible without having to stitch together information from other docs / sources. 

> I'm happy to submit a PR to make these fixes, but I wanted to create this discussion first to make sure I set it up as intended.

I appreciate it @bpmct, the feedback is well received and we would greatly appreciate any contribution. 

We should also make follow up issues to make sure the same issues do not exist with the other quick starts.
